LoopKitUI/Views/ScheduleEditor.swift

```diff
@@ -0,0 +1,281 @@
+
+                    if self.scheduleItems.count == 1 {
+                        self.tableDeletionState = .disabled
+                    }
+                }
+            }
+        ) {
+            ScheduleItemView(
+                time: item.startTime,
+                isEditing: isEditing(index),
+                valueContent: {
+                    valueContent(item.value, isEditing(index).wrappedValue)
+                },
+                expandedContent: {
+                    ScheduleItemPicker(
+                        item: $scheduleItems[index],
+                        isTimeSelectable: { self.isTimeSelectable($0, at: index) },
+                        valuePicker: { self.valuePicker(self.$scheduleItems[index]) }
+                    )
+                }
+            )
+        }
+    }
+
+    private func isEditing(_ index: Int) -> Binding<Bool> {
+        Binding(
+            get: { index == self.editingIndex },
+            set: { isNowEditing in
+                self.editingIndex = isNowEditing ? index : nil
+            }
+        )
+    }
+
+    private func isTimeSelectable(_ time: TimeInterval, at index: Int) -> Bool {
+        if index == scheduleItems.startIndex {
+            return time == .hours(0)
+        }
+
+        let priorTime = scheduleItems[index - 1].startTime
+        guard time > priorTime else {
+            return false
+        }
+
+        if index < scheduleItems.endIndex - 1 {
+            let nextTime = scheduleItems[index + 1].startTime
+            guard time < nextTime else {
+                return false
+            }
+        }
+
+        return true
+    }
+
+    private var initialNewScheduleItem: RepeatingScheduleValue<Value> {
+        assert(scheduleItems.count <= scheduleItemLimit)
+
+        if scheduleItems.last!.startTime == .hours(23.5) {
+            let firstItemFollowedByOpening = scheduleItems.adjacentPairs().first(where: { item, next in
+                next.startTime - item.startTime > .minutes(30)
+            })!.0
+            return RepeatingScheduleValue(
+                startTime: firstItemFollowedByOpening.startTime + .minutes(30),
+                value: firstItemFollowedByOpening.value
+            )
+        } else {
+            return RepeatingScheduleValue(
+                startTime: scheduleItems.last!.startTime + .minutes(30),
+                value: scheduleItems.last!.value
+            )
+        }
+    }
+
+    private var trailingNavigationItems: some View {
+        // TODO: SwiftUI's alignment of these buttons in the navigation bar is a little funky.
+        // Tapping 'Edit' then 'Done' can shift '+' slightly.
+        HStack(spacing: 24) {
+            if tableDeletionState == .disabled {
+                editButton
+            } else {
+                doneButton
+            }
+
+            addButton
+        }
+    }
+
+    var cancelButton: some View {
+        Button(action: dismiss, label: { Text("Cancel") })
```

Comment:
Not framework localized.

Reply:
None of our LoopKit SwiftUI strings are framework localized, right? https://tidepool.atlassian.net/browse/LOOP-1251?focusedCommentId=19865&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-19865

Reply:
I can add the missing localization comment, though the string literal here is inferred to be LocalizedStringKey, so it isn't a raw string.

Reply:
Ok, yeah, that's fine if we defer it for that ticket.